### PR TITLE
Top3 Challenge view on front page

### PR DESCRIPTION
**Pull request by rl-86**
Skärmbild (82)
Skärmbild (81)

* Added top3 view on front page
* Converted rating value to stars
* Finished styling the cards
* Made some corrections
* Cleanup unused code

### CODE REVIEW

**Review by simonpizevski — APPROVED** [review]
Snyggt jobbat med top3 view! Ser riktigt bra ut

File: api_fetch.js [hunk 1]
```diff
@@ -17,48 +19,70 @@ export default class Challenge {
...
+
+      for (let i = 0; i < fullStars; i++) {
+        ratingContainer.innerHTML += fullStar;
+      }
+
+      if (hasHalfStar) {
+        ratingContainer.innerHTML += halfStar;
+      }
+
+      const remainingStars = hasHalfStar ? 5 - fullStars - 1 : 5 - fullStars;
```
Comment (richardolsson) [inline api_fetch.js:54]: I believe this could have been simplified with `const remainingStars = Math.floor(5 - fullStars);`.

File: api_fetch.js [hunk 2]
```diff
@@ -131,12 +148,36 @@ class ChallengeListView {
...
+    for (let i = 0; i < 3; i++) {
+      const challenge = sorted[i];
+      const element = challenge.render();
+      container.append(element);
+    }
+  }
+}
+
+// Look for which page is showing and render all challenges or top3
+const currentPage = document.body.dataset.page;
```
Comment (richardolsson) [inline api_fetch.js:168]: Nice use of `data-` attributes and `dataset`! For things like this, the HTML `class` attribute is often used, but this is valid as well and perhaps even better if the value is only relevant to JS (not CSS).

File: api_fetch.js [hunk 3]
```diff
@@ -328,4 +369,4 @@ function searchTags(tag) {
       noMatchError.innerHTML = "No matching challenges";
     }
   }
-}
+}
```
Comment (richardolsson) [inline api_fetch.js:372]: Here, the only change that has happend is related to whitespace. In the pre-existing version, there was a final newline character (i.e. an "empty line") after the last curly brace. In your updated version, there is none.

Different editors (and differently configured instances of the same editor) will handle things like this differently. In a team, it's always good to agree on a specific practice and configure editors (or other tools) to stick to that practice.

**Review by richardolsson — COMMENTED** [review]
Nice work! I've left some comments that can hopefully be useful for the future.

**Pull request merged**